src/setup.jl

```diff
@@ -243,22 +243,43 @@ function regenerate_gaproot(gaproot_mutable)
         # create a `pkg` symlink to the GAP packages artifact
         force_symlink(artifact"gap_packages", "pkg")
 
-        ##
-        ## Create Project.toml & Manifest.toml for use by gap.sh
-        ##
-        @info "Generating custom Julia project ..."
-        relative_pkgdir = joinpath("..", "..")
-        @assert abspath(gaproot_mutable, relative_pkgdir) == gaproot_gapjl
-        run(pipeline(`$(Base.julia_cmd()) --startup-file=no --project=$(gaproot_mutable) -e "using Pkg; Pkg.develop(PackageSpec(path=\"$(relative_pkgdir)\"))"`))
+    end # cd(gaproot_mutable)
+
+    ##
+    ## Compile JuliaInterface
+    ##
+    @info "Compiling JuliaInterface ..."
+
+    cd(joinpath(gaproot_gapjl, "pkg", "JuliaInterface")) do
+        run(pipeline(`./configure $gaproot_mutable`, stdout="build.log"))
+        run(pipeline(`make V=1 -j$(Sys.CPU_THREADS)`, stdout="build.log", append=true))
+    end
+
+    ##
+    ## Create custom gap.sh
+    ##
+    create_gap_sh(joinpath(gaproot_mutable, "bin"))
+
+    return gaproot_mutable
+end # function
 
-    end # cd
+
+function create_gap_sh(dstdir)
```

Comment:
Ultimately, a docstring needs to be added and perhaps a better name be selected.

Also, if users are supposed to call this, perhaps some sanity tests should be performed? 

Should we require `dstdir` to exist? Or do the opposite and demand that it must not exist? Or create it if missing?